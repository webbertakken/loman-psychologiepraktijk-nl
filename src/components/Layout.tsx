interface Props {
  children: JSX.Element
}

const Layout = ({ children }: Props): JSX.Element => {
  return (
    <div>
      <main>{children}</main>
    </div>
  )
}

export default Layout
