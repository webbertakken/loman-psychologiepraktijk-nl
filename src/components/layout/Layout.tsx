import MenuLinks from './menu/MenuLinks'

interface Props {
  children: JSX.Element
  menu?: Record<string, any>[]
}

const Layout = ({ children, menu }: Props): JSX.Element => {
  return (
    <div>
      <header>
        <MenuLinks menu={menu} />
      </header>
      <main>{children}</main>
    </div>
  )
}

export default Layout
