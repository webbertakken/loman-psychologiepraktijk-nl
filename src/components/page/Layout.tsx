import NavBar from './navbar/NavBar';
import Banner from './Banner';
import { BannerNotificationEntry } from '../../types/banner';
import Footer from './footer/Footer';

interface Props {
  children: JSX.Element;
  banner?: BannerNotificationEntry;
}

const Layout = ({ children }: Props): JSX.Element => {
  return (
    <div className="font-serif min-h-screen flex flex-col">
      <Banner />
      <NavBar />
      <main>{children}</main>
      <Footer />
    </div>
  );
};

export default Layout;
