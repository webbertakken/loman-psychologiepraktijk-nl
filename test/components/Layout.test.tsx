import React from 'react';
import { render, screen, fireEvent } from '../testUtils';
import Layout from '../../src/components/page/layout/Layout';

describe('Layout', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Layout>{null}</Layout>, {});
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders renders children', () => {
    render(
      <Layout>
        <button onClick={() => alert('button is clickable')}>Test Button</button>
      </Layout>,
      {},
    );
    window.alert = jest.fn();
    fireEvent.click(screen.getByText('Test Button'));
    expect(window.alert).toHaveBeenCalledWith('button is clickable');
  });
});
