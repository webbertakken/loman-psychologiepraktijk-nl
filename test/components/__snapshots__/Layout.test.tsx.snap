// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="font-serif min-h-screen flex flex-col"
  >
    <section
      class="hidden md:block relative w-full px-8 text-gray-300 body-font z-10"
    >
      <div
        class="container flex flex-col flex-wrap items-center justify-between py-5 mx-auto md:flex-row max-w-7xl"
      >
        <a
          class="z-10"
          href="/"
        >
          <test-file-stub
            classname="h-14 select-none cursor-pointer"
          />
        </a>
        <nav
          class="top-0 left-0 z-0 flex items-center justify-center h-full py-5 -ml-0 space-x-5 text-base md:-ml-5 md:py-0"
        />
      </div>
    </section>
    <main />
    <section
      class="py-10 bg-black mt-auto"
    >
      <div
        class="px-10 mx-auto max-w-7xl"
      >
        <div
          class="flex flex-col justify-between text-center md:flex-row"
        >
          <p
            class="order-last text-sm leading-tight text-gray-500 md:order-first"
          >
            <strong>
              Loman Psychologiepraktijk
            </strong>
            <strong>
               | 
            </strong>
            <span>
              Alle rechten voorbehouden
            </span>
          </p>
          <nav
            class="flex flex-row justify-center pb-3 -ml-4 -mr-4 text-sm"
          />
        </div>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
